webserver/cors/src/main/java/io/helidon/webserver/cors/Aggregator.java

```diff
@@ -0,0 +1,307 @@
+
+import static io.helidon.webserver.cors.CORSSupportHelper.normalize;
+
+/**
+ * Collects CORS set-up information from various sources and looks up the relevant CORS information given a request's path.
+ * <p>
+ *    The caller can build up the cross-config information over multiple invocations of the exposed methods. The behavior is that
+ *    of a {@link LinkedHashMap}:
+ *    <ul>
+ *        <li>when <em>storing</em> cross-config information, the <em>latest</em> invocation that specifies the same path
+ *        expression overwrites any preceding settings for the same path expression, and</li>
+ *        <li>when <em>matching</em> against a request's path, the code checks the path matchers <em>in the order
+ *        they were added</em> to the aggregator, whether by {@link #mappedConfig} or {@link #addCrossOrigin} or the {@link Setter}
+ *        methods.
+ *    </ul>
+ * </p>
+ * <p>
+ *     The {@code Setter} methods affect the so-called "pathless" entry. Those methods have no explicit path, so we record
+ *     their settings in an entry with path expression {@value #PATHLESS_KEY} which matches everything.
+ * </p>
+ * <p>
+ *     If the developer uses the {@link #mappedConfig} or {@link #addCrossOrigin} methods <em>along with</em> the {@code Setter}
+ *     methods, the results are predictable but might be confusing. The {@code config} and {@code addCrossOrigin} methods
+ *     <em>overwrite</em> any entry with the same path expression, whereas the {@code Setter} methods <em>update</em> an existing
+ *     entry with path {@value #PATHLESS_KEY}, creating one if needed. So, if the config or an {@code addCrossOrigin}
+ *     invocation sets values for that same path expression then results can be surprising.
+ *     path
+ * </p>
+ *
+ */
+class Aggregator implements Setter<Aggregator> {
+
+    // Key value for the map corresponding to the cross-origin config managed by the {@link Setter} methods
+    static final String PATHLESS_KEY = "{+}";
+
+    // Records paths and configs added via addCrossOriginConfig
+    private final Map<String, CrossOriginConfigMatchable> crossOriginConfigMatchables = new LinkedHashMap<>();
+
+    private boolean isEnabled = true;
+
+    /**
+     * Factory method.
+     *
+     * @return new CrossOriginConfigAggregatpr
+     */
+    static Aggregator create() {
+        return new Aggregator();
+    }
+
+    private Aggregator() {
+    }
+
+    /**
+     * Reports whether the sources of CORS information have left CORS enabled or not. If there has been an explicit setting,
+     * use the most recent. Otherwise
+     *
+     * @return if CORS processing should be done
+     */
+    public boolean isEnabled() {
+        return isEnabled;
+    }
+
+    /**
+     * Add cross-origin information from a {@link Config} node.
+     *
+     * @param config {@code Config} node containing
+     * @return updated builder
+     */
+    public Aggregator mappedConfig(Config config) {
```

Comment:
Class is package local, `public` methods should not exist here

Reply:
I have changed the ones I can. Some are required by an interface and are therefore remain `public`.